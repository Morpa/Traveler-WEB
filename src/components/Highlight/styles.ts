import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    width: 16rem;
    height: 27rem;
    background: ${theme.colors.primary};
    border: solid 1px ${theme.colors.secondary};
    border-radius: 2rem;
  `}
`
export const Icon = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    height: 10rem;

    margin-right: ${theme.spacings.xxsmall};
    padding: ${theme.spacings.medium};

    & > svg {
      width: 5.2rem;
      color: ${theme.colors.orange};
    }
  `}
`
export const Info = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.medium};
    border-top: solid 1px ${theme.colors.secondary};

    > h1 {
      color: ${theme.colors.titleColor};
      font-size: 4rem;
      font-weight: ${theme.font.bold};
      margin-bottom: 1rem;
    }

    > span {
      color: ${theme.colors.textColor};
      font-size: ${theme.font.sizes.xsmall};
      font-weight: ${theme.font.light};
      line-height: 2.2rem;
    }
  `}
`
