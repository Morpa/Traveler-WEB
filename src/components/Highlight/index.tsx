import * as S from './styles'

export type HighlightProps = {
  icon: React.ReactNode
  count: number
  title: string
}

const Highlight = ({ icon, count, title }: HighlightProps) => (
  <S.Wrapper>
    <S.Icon aria-label="Icon">{icon}</S.Icon>
    <S.Info>
      <h1 aria-label="Quantidade do cartão">{count}</h1>
      <span aria-label="Título do cartão">{title}</span>
    </S.Info>
  </S.Wrapper>
)

export default Highlight
