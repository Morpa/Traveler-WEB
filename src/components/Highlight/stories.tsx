import { Story, Meta } from '@storybook/react/types-6-0'
import { Camera } from '@styled-icons/bootstrap/Camera'
import Highlight, { HighlightProps } from '.'

export default {
  title: 'Highlight',
  component: Highlight,
  argTypes: {
    icon: {
      type: ''
    }
  }
} as Meta

export const Default: Story<HighlightProps> = (args) => <Highlight {...args} />
Default.args = {
  icon: <Camera />,
  title: 'Pontos Tur√≠sticos',
  count: 65
}
