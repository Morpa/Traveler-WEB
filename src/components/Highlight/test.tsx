import { screen } from '@testing-library/react'
import { Camera } from '@styled-icons/bootstrap/Camera'
import { renderWithTheme } from 'utils/tests/helpers'

import Highlight from '.'

const props = {
  icon: <Camera />,
  title: 'Pontos Turísticos',
  count: 65
}

describe('<Highlight />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(screen.getByLabelText(/título do cartão/i)).toBeInTheDocument()

    expect(screen.getByLabelText(/quantidade do cartão/i)).toBeInTheDocument()

    expect(screen.getByLabelText(/icon/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
