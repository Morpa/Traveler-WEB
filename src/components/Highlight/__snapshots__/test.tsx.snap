// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render the heading 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 16rem;
  height: 27rem;
  background: #FFFFFF;
  border: solid 1px #DCE2E6;
  border-radius: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 10rem;
  margin-right: 0.8rem;
  padding: 3.2rem;
}

.c1 > svg {
  width: 5.2rem;
  color: #F25D27;
}

.c3 {
  padding: 3.2rem;
  border-top: solid 1px #DCE2E6;
}

.c3 > h1 {
  color: #123952;
  font-size: 4rem;
  font-weight: 600;
  margin-bottom: 1rem;
}

.c3 > span {
  color: #617480;
  font-size: 1.2rem;
  font-weight: 300;
  line-height: 2.2rem;
}

<div
  class="c0"
>
  <div
    aria-label="Icon"
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 12a1 1 0 01-1 1H2a1 1 0 01-1-1V6a1 1 0 011-1h1.172a3 3 0 002.12-.879l.83-.828A1 1 0 016.827 3h2.344a1 1 0 01.707.293l.828.828A3 3 0 0012.828 5H14a1 1 0 011 1v6zM2 4a2 2 0 00-2 2v6a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1.172a2 2 0 01-1.414-.586l-.828-.828A2 2 0 009.172 2H6.828a2 2 0 00-1.414.586l-.828.828A2 2 0 013.172 4H2z"
      />
      <path
        d="M8 11a2.5 2.5 0 110-5 2.5 2.5 0 010 5zm0 1a3.5 3.5 0 100-7 3.5 3.5 0 000 7zM3 6.5a.5.5 0 11-1 0 .5.5 0 011 0z"
      />
    </svg>
  </div>
  <div
    class="c3"
  >
    <h1
      aria-label="Quantidade do cartão"
    >
      65
    </h1>
    <span
      aria-label="Título do cartão"
    >
      Pontos Turísticos
    </span>
  </div>
</div>
`
